tests/test_examples.py

```diff
@@ -0,0 +1,386 @@
+        retcode = p.poll()
+        if check and retcode:
+            raise subprocess.CalledProcessError(retcode, cmd, output=stdout, stderr=stderr)
+        return subprocess.CompletedProcess(cmd, retcode, stdout, stderr)
+    except subprocess.TimeoutExpired:
+        p.kill()
+        stdout, stderr = p.communicate()
+        raise
+    finally:
+        if stdout:
+            print(stdout)
+        if stderr:
+            print(stderr, file=sys.stderr)
+        print("Took", timedelta(seconds=time.time() - t0))
+
+
+def _run_installer_exe(installer, install_dir, installer_input=None):
+    """
+    NSIS manual:
+    > /D sets the default installation directory ($INSTDIR), overriding InstallDir
+    > and InstallDirRegKey. It must be the last parameter used in the command line
+    > and must not contain any quotes, even if the path contains spaces. Only
+    > absolute paths are supported.
+    Since subprocess.Popen WILL escape the spaces with quotes, we need to provide
+    them as separate arguments. We don't care about multiple spaces collapsing into
+    one, since the point is to just have spaces in the installation path -- one
+    would be enough too :)
+    This is why we have this weird .split() thingy down there in `/D=...`.
+    """
+    if not sys.platform.startswith("win"):
+        raise ValueError("Can only run .exe installers on Windows")
+    if "NSIS_USING_LOG_BUILD" not in os.environ:
+        warnings.warn(
+            "Windows installers are tested with NSIS in silent mode, which does "
+            "not report errors on exit. You should use logging-enabled NSIS builds "
+            "to generate an 'install.log' file this script will search for errors "
+            "after completion."
+        )
+    cmd = ["cmd.exe", "/c", "start", "/wait", installer, "/S", *f"/D={install_dir}".split()]
+    _execute(cmd)
+
+    # Windows installers won't raise exit codes so we need to check the log file
+    error_lines = []
+    try:
+        log_is_empty = True
+        with open(os.path.join(install_dir, "install.log"), encoding="utf-16-le") as f:
+            for line in f:
+                log_is_empty = False
+                if ":error:" in line:
+                    error_lines.append(line)
+        if log_is_empty:
+            error_lines.append("Logfile was unexpectedly empty!")
+    except Exception as exc:
+        error_lines.append(
+            f"Could not read logs! {exc.__class__.__name__}: {exc}\n"
+            "This usually means that the destination folder could not be created.\n"
+            "Possible causes: permissions, non-supported characters, long paths...\n"
+            "Consider setting 'check_path_spaces' and 'check_path_length' to 'False'."
+        )
+    if error_lines:
+        raise AssertionError("\n".join(error_lines))
+
+
+def _run_uninstaller_exe(install_dir):
+    # Now test the uninstallers
+    if " " in str(install_dir):
```

Comment:
Better to raise an exception here? Send safer than silently allowing the test to pass (assuming that's the outcome of short circuiting here)

Reply:
That's a known issue we couldn't fix in CI (works locally), but I agree we could at least add a warning!